File: src/network/packet/impl/ProxyUnregisterServerPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl;

use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\ClientboundPacket;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\util\PacketData;

final class ProxyUnregisterServerPacket extends CloudPacket implements ClientboundPacket {

    public function __construct(private readonly string $serverName = "") {}

    public function handle(ServerClient $client): void {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->writeAll($this->serverName);
    }

    public function decodePayload(PacketData $packetData): void {}

    public function getServerName(): string {
        return $this->serverName;
    }

    public static function create(string $serverName): self {
        return new self($serverName);
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/network/packet/impl/ProxyUnregisterServerPacket.java
package de.pocketcloud.cloud.network.packet.impl;

import de.pocketcloud.cloud.network.client.ServerClient;
import de.pocketcloud.network.packet.ClientboundPacket;
import de.pocketcloud.cloud.network.packet.CloudPacket;
import de.pocketcloud.network.packet.data.PacketData;
import de.pocketcloud.cloud.server.CloudServer;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

@NoArgsConstructor
@Getter
public final class ProxyUnregisterServerPacket extends CloudPacket implements ClientboundPacket {

    private String serverName;

    public ProxyUnregisterServerPacket(String serverName) {
        this.serverName = serverName != null ? serverName : "";
    }

    @Override
    public void handle(@NotNull ServerClient client) {}

    @Override
    public void encodePayload(PacketData packetData) {
        packetData.writeAll(serverName);
    }

    @Override
    public void decodePayload(PacketData packetData) {}

    public static ProxyUnregisterServerPacket create(String serverName) {
        return new ProxyUnregisterServerPacket(serverName);
    }

    public static ProxyUnregisterServerPacket create(CloudServer server) {
        return new ProxyUnregisterServerPacket(server.name());
    }
}
